storage/data/consumer.go

```diff
@@ -1,13 +1,18 @@
 package data
 
-import "net/url"
+import (
+	"net/url"
+
+	"github.com/newscred/webhook-broker/config"
+)
 
 // Consumer is the object that producer broadcasts to and consumer consumes from
 type Consumer struct {
 	MessageStakeholder
 	ConsumerID    string
 	CallbackURL   string
 	ConsumingFrom *Channel
+	Type          config.ConsumerType
```

Comment:
The `ConsumerType` is part of the consumer's data model, so IMO it makes more sense to define the type here

Reply:
Yes bhaia. But I am also using this https://github.com/newscred/webhook-broker/pull/140/files#diff-2f690a05792c7f0a5738fb4e4bdc87dc2bdb8f188a227b88fccf271e09adb44cR82 . That is why I am confused about where to place it. I tried to explain the dilemma here https://github.com/newscred/webhook-broker/pull/140#discussion_r1104206458

Reply:
If you move the validation to `NewConsumer`, you can keep the ConsumerType in the config a string.
IMO this makes sense, since we should validate whenever we initialize the consumer anyway, so the validation in the config feels a bit redundant.